### DISCUSSION TITLE
Setting Default Password for web server in .Bin file and forcing password change at first boot

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#12302] [Q&A] [answered]
It is my understanding that for security reasons it is recommended that Tasmota web interface should be protected with a password for minimum security. ( I also understand that recommendation is to turn off the web server during routine use after necessary configuration changes have been made and saved.) 

Is it possible to set a default password for web server in .bin file?
If yes, then is it possible to force change this password when the device boots for the first time and prompts for WiFi credentials? From my research so far it looks like that it is not possible to do both of the above. That leads me to my third question:
Can above features be implemented in future versions of Tasmota? 

Regards,

Navi

**Comment by sfromis** [comment, marked as answer]
If you talk about actual security, that password is so weak that it does not count much, and not much of a recommendation actually. More of like a thing to not have the candy cupboard too easy to open.  If you want a bit better security, you can run the device in admin mode (instead of user), where you can only unlock it via serial or MQTT. Still no "strong" security there, of course. You could also compile a binary without web server at all, to make it a bit harder to play around....

Using `#define WEB_PASSWORD` you can have a password in the binary, but no option for forced change.

**Reply by sfromis** [reply]
As you may know, the "S" in "IoT" stands for security ;-)

**Comment by NsinghP** [comment]
LOL, I really miss the S, but Siot doesn’t have that satisfying tone to it,
I guess we’ll just have to live without S for the time being :) by the way
is security situation any better on ESP32 side?

**Reply by sfromis** [reply]
Tasmota is not trying to add "hard" security features there too.

**Reply by Jason2866** [reply]
The idea of Tasmota is to use it in a local secured Wifi environment.
Doing this way you have a no more a IoT device just oT ;-)

**Reply by NsinghP** [reply]
> The idea of Tasmota is to use it in a local secured Wifi environment.
> Doing this way you have a no more a IoT device just oT ;-)

Ha ha, I think it still is an IoT device with a slightly different interpretation of the I. “Intranet of Things“ device anyone?

**Reply by NsinghP** [reply]
Here is a good article on IoT vs. IoT:

*https://tele2iot.com/blog/internet-of-things-vs-intranet-of-things/
